ui/js/compareperf.js

```diff
@@ -5,12 +5,23 @@
 "use strict";
 
 perf.controller('CompareChooserCtrl', [
-  '$state', '$stateParams', '$scope', 'ThRepositoryModel', 'ThResultSetModel',
-  function CompareChooserCtrl($state, $stateParams, $scope,
-                              ThRepositoryModel, ThResultSetModel) {
+  'phCompareDefaultNewRepo', 'phCompareDefaultOriginalRepo', '$state', '$stateParams',
+   '$scope', 'ThRepositoryModel', 'ThResultSetModel',
+  function CompareChooserCtrl(phCompareDefaultNewRepo, phCompareDefaultOriginalRepo, $state, $stateParams,
+                              $scope, ThRepositoryModel, ThResultSetModel) {
     ThRepositoryModel.get_list().success(function(projects) {
       $scope.projects = projects;
-      $scope.originalProject = $scope.newProject = projects[0];
+      $scope.originalProject = getProject(phCompareDefaultOriginalRepo);
+      $scope.newProject = getProject(phCompareDefaultNewRepo);
+
+      function getProject(repoName) {
+        var ret = ThRepositoryModel.getRepo(repoName);
```

Comment:
getRepo implies that we've successfully called load() on ThRepositoryModel, which we haven't. Also, there will be a mismatch between the object returned by getRepo and the projects list you just got. Just replacing this line with `var ret = _.findWhere(projects, {name: repoName});` should do the trick.